src/ZeroCrossImpl.java

```diff
@@ -0,0 +1,177 @@
+import java.util.Scanner;
+
+public class ZeroCrossImpl implements ZeroCross {
+
+    int whoPlays;
+    public int firstPlayer;
+    public int secondPlayer;
+
+
+    @Override
+    public void run() {
+
+
+        Scanner scanner = new Scanner(System.in);
```

Comment:
Better use type of reading data out of target class in case if we want to use another source.